framework/log/Dispatcher.php

```diff
@@ -80,7 +80,7 @@ public function __construct($config = [])
     {
         // ensure logger gets set before any other config option
         if (isset($config['logger'])) {
-            $this->setLogger($config['logger']);
+            $this->setLogger(Yii::createObject($config['logger']));
```

Comment:
Looks like you have to check if `$config['logger']` is actually an array, otherwise `createObject()` will complain (see failing tests). This might work:
`$this->setLogger(is_array($config['logger']) ? Yii::createObject($config['logger']) : $config['logger']);`